### ISSUE TITLE
Warn about indented blocks that contain a `pass` statement and have further statements, just like with `return`

### ISSUE THREAD

**Issue by whisperity** [microsoft/pyright#11240]
## Is your feature request related to a problem? Please describe.

Not exactly a problem, but in one of my recent deliverables, I accidentally created code that contains the following (minified) bogus behaviour:

```py
try:
  cfg[key] = reader.get_as_int(key)
  continue
except ValueError:
  # Intentionally left empty.
  pass
try:
  cfg[key] = reader.get_as_bool(key)
  continue
except ValueError:
  # Intentionally left empty.
  pass

  # ???
  value = reader.get(key, raw_string=True)
if isinstance(value, str):
  if ';' in v:
    value = v.split(';')
# value is either a list now, or if it wasn't ;-separated, remained a str
if isinstance(value, list):
  cfg[key] = {}
  for i, e in enumerate(value):
    cfg[key][str(i)] = e
else:
  cfg[key] = value
```

Notice how `value = reader.get(key, raw_string=True)` only gets executed if `get_as_bool` throws `ValueError`! Technically, this is not an issue, because in this specific control-flow, we would always hit that `except ValueError:` block in case the string in the `reader` was `;`-separated.

In general, however, an identation block that contains an explicit `pass` (indicating that the developer **intentionally left it empty** and put the `pass` there just to comply with the grammar because a truly "empty" block is syntactically invalid) followed by some actual statements on the same indentation level should indicate a code smell!

## Describe the solution you’d like

Similarly to how statements after a **`return`** in a block are highlighted as never possible to execute:

```py
try:
  statements()
except ErrorType:
  return

  # Code is structurally unreachable (Pyright)
  some_other_statement()
```

Pyright should highlight as an issue for **all sorts of blocks** if:

1. A `pass` statement is preceded by some statements — mentioning that the `pass` is superfluous.
2. A `pass` statement dominates some other statements — mentioning that this is misleading, and perhaps the intention was to have the dominated statement unintended.

```py
if condition():
  stmt1()
  pass    # Superfluous 'pass' has no effect.
  stmt2()
```

```py
try:
  stmt()
except:
  pass

  # Potentially misleading / unintended identation, stmt2() only executes in the 'except' block.
  stmt2()
```

**Comment by erictraut** [comment]
Thanks for the suggestion, but these features fall more into the category of a linter. They're not the domain of a type checker. We've generally tried to steer clear of features that a linter (which typically has no type evaluation capabilities) can implement. The most popular linters likely already have checks for these conditions. If not, I recommend that you file a feature request with your favorite linter.
